### PR TITLE
[remoteopenhab] Consider the remote item pattern formatter

### LINKED ISSUE #9645: [remoteopenhab] Considering the remote item pattern formatter
Currently, the remote openHAB binding considers whether the remote item is read-only or not; it also considers the state options from the remote item. But it does not yet consider the remote item pattern formatter as pattern for the local channel (and so linked local item).
One of the consequence is that for example if your remote item is set to display a unit, your local item will not have this unit displayed by default. You will have to setup it additionally in the main server.

The enhancement will consist in applying the same pattern formatter as the one defined for the remote item.
Technically, this will imply more or less to create one channel type for each remote item, rather than using few generic channel types).

I will try to implement this enhancement.

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #9645

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/RemoteopenhabChannelTypeProvider.java [hunk 1]
```diff
@@ -50,11 +56,51 @@
...
+    public void addChannelType(String itemType, ChannelType channelType) {
+        channelTypes.add(channelType);
+        List<ChannelType> channelTypesForItemType = channelTypesForItemTypes.get(itemType);
+        if (channelTypesForItemType == null) {
+            channelTypesForItemTypes.put(itemType, new CopyOnWriteArrayList<>());
+            channelTypesForItemType = channelTypesForItemTypes.get(itemType);
+        }
+        if (channelTypesForItemType != null) {
+            channelTypesForItemType.add(channelType);
+        }
```
Comment (cpmeister) [inline bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/RemoteopenhabChannelTypeProvider.java:96]:
```suggestion
        List<ChannelType> channelTypesForItemType = channelTypesForItemTypes.computeIfAbsent(itemType, type -> new CopyOnWriteArrayList<>());
        if (channelTypesForItemType != null) {
            channelTypesForItemType.add(channelType);
        }
```

**Review by cpmeister — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/RemoteopenhabChannelTypeProvider.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/RemoteopenhabChannelTypeProvider.java]: Good to know. Applied

**Comment by lolodomo** [comment]
@cpmeister : do you think I should remove `List<ChannelType> channelTypes` ? I kept it only to have the method `getChannelTypes` simple (just returning it). But I could certainly build the returned value for `getChannelTypes` with my new hashmap.

**Comment by cpmeister** [comment]
I don't think it matters all that much. As long as the code is simple and performant.

**Review by cpmeister — APPROVED** [review]

**Pull request merged**